lib/puppet/indirector/file_server.rb

```diff
@@ -51,6 +52,7 @@ def search(request)
     Puppet::FileServing::Fileset.merge(*filesets).collect do |file, base_path|
       inst = model.new(base_path, :relative_path => file)
       inst.links = request.options[:links] if request.options[:links]
+      inst.checksum_type = request.options[:checksum_type] if request.options[:checksum_type]
       inst.collect
```

Comment:
I'll open a new ticket. Adding it doesn't break any tests, so we'll need some test coverage for it.